pkg/plugin/group/rollingupdate.go

```diff
@@ -149,37 +162,37 @@ func (r *rollingupdate) Run(pollInterval time.Duration) error {
 		return err
 	}
 
+	// First determine if any new instances should be created
 	desired, _ := desiredAndUndesiredInstances(instances, r.updatingTo)
 	expectedNewInstances := len(desired)
+	log.Info("RollingUpdate-Run", "expectedNewInstances", expectedNewInstances)
 
 	for {
-		err := r.waitUntilQuiesced(
-			pollInterval,
-			minInt(expectedNewInstances, int(r.updatingTo.config.Allocation.Size)))
+		// Wait until any new nodes are healthy
+		err := r.waitUntilQuiesced(pollInterval, minInt(expectedNewInstances, r.desiredSize))
```

Comment:
`r.desiredSize` is referenced here only to determine the min count to wait on.  `r.desiredSize` is initialized to be either `len(config.Allocations.LogicalIDs)` or `int(config.Allocations.Size)`...    Do you think it'd be more direct to just determine the `desiredSize` here... something like

```
desiredSize := len(r.updatingTo.config.Allocation.LogicalIDs)
if desiredSize == 0 {
   desiredSize = int(r.updatingTo.config.Allocation.Size)
}
err := r.waitingUntilQuiesced(pollInterval, minInt(expectedNewInstances, desiredSize))
...
```

Then we don't even need the extra field in `r`?   seems like this isn't something the user can set anyway..

Reply:
Yeah, this makes sense and then we don't need to have the struct value.

Reply:
I think computing it there would make the logic easier to follow and would address the question I had in `scaler.go` - about mutating that field when the group size is changed...